### PR TITLE
fix: `ContentPresenter` content alignment

### LINKED ISSUE #9361: [BUTTON] HorizontalAlignment of content wrongly applied when text
### Current behavior

Take this XAML:

``` xml
<StackPanel Spacing="20">
  <Button HorizontalAlignment="Stretch">123</Button>
  <Button HorizontalAlignment="Stretch" ><TextBlock Text="123" HorizontalAlignment="Stretch" /></Button>
  <Button HorizontalAlignment="Stretch" ><TextBlock Text="123" /></Button>
</StackPanel>
```

It's currently producing 3 different results, where the 2nd is left-aligned.

### Expected behavior

Put the same code in XAML Studio and those 3 text are centered.

### How to reproduce it (as minimally and precisely as possible)

https://playground.platform.uno/#adfc5e7a

### Workaround

_No response_

### Works on UWP/WinUI

_No response_

### Environment

_No response_

### NuGet package version(s)

Uno 4.4

### Affected platforms

WebAssembly

### IDE

_No response_

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): closes #9361

## PR Type

What kind of change does this PR introduce?

## What is the current behavior?

Previously content alignement properties on ContentPresenter were overriding the alignment of the child, which did not work properly in many cases (especially when the alignment was directly set on the content element). Instead ContentPresenter should arrange the content based on the content alignment properties.

## What is the new behavior?

Content presenter alignment behaves correctly.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [x] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Comment by MartinZikmund** [comment]
TODO runtime tests for all edge cases

File: src/Uno.UI/UI/Xaml/Controls/ContentPresenter/ContentPresenter.cs [hunk 1]
```diff
@@ -1089,19 +1010,88 @@ protected override Size ArrangeOverride(Size finalSize)
...
 					finalSize.Height - padding.Top - padding.Bottom - borderThickness.Top - borderThickness.Bottom
 				);
 
-				base.ArrangeElement(child, finalRect);
+				var availableSize = new Size(innerRect.Width, innerRect.Height);
+
+				// Using GetElementDesiredSize to properly handle native controls.
+				var desiredSize = GetElementDesiredSize(child);
+
+				var contentWidth = HorizontalContentAlignment == HorizontalAlignment.Stretch ?
```
Comment (Xiaoy312) [inline src/Uno.UI/UI/Xaml/Controls/ContentPresenter/ContentPresenter.cs:1025]: the generic implementation of arranging given the content and available size can be move to LayoutHelper
it is something that is used frequently, yet the implementation is never reused

Comment (MartinZikmund) [reply src/Uno.UI/UI/Xaml/Controls/ContentPresenter/ContentPresenter.cs:1025]: Will refactor and move it there

**Review by dr1rrb — APPROVED** [review]

**Comment by carldebilly** [comment]
Please check the following screenshots in your branch:
* https://dev.azure.com/uno-platform/Uno%20Platform/_build/results?buildId=51231&view=ms.vss-test-web.build-test-results-tab&runId=1898514&resultId=100048&paneView=attachments
* https://dev.azure.com/uno-platform/Uno%20Platform/_build/results?buildId=51231&view=ms.vss-test-web.build-test-results-tab&runId=1898514&resultId=100181&paneView=attachments
* + a log of images seems absent in screenshot compares...  are they broken?

File: src/Uno.UI/UI/Xaml/Controls/ContentPresenter/ContentPresenter.cs (hunk 1 above)
Comment (MartinZikmund) [reply src/Uno.UI/UI/Xaml/Controls/ContentPresenter/ContentPresenter.cs:1025]: In the end I decided against moving it, I can explain over Teams :-) .

**Comment by MartinZikmund** [comment]
@carldebilly checked both issues, but it seems it was just that the screenshots were taken too early (the image didn't load yet, as it is external; and the date picker flyout shortly flashes without content - but that is happening even on main branch)

**Comment by jeromelaban** [comment]
A few notable changes :
- appbar overflow "dots" positioning: https://dev.azure.com/uno-platform/Uno%20Platform/_build/results?buildId=51231&view=ms.vss-test-web.build-test-results-tab&runId=1898514&resultId=100870&paneView=attachments
- "Content" button position: https://dev.azure.com/uno-platform/Uno%20Platform/_build/results?buildId=51231&view=ms.vss-test-web.build-test-results-tab&runId=1898498&resultId=100327&paneView=attachments

**Comment by MartinZikmund** [comment]
@jeromelaban should be correct, from MUX test app:

image

The dots positioning also seems closer to WinUI now 👍
